### PR TITLE
fix(aws-lambda): fetch ecs environment variable failure caused by the phase change

### PR DESCRIPTION
**Pull request by windmgc**
### Summary

This PR fixes a bug that the aws-lambda plugin failed to get environment variables in ECS which leads to unable to fetch ECS task role credentials.

This bug is introduced since #8900, in which the moment of the step `require "kong.plugins.aws-lambda.iam-ecs-credentials"` is changed from happening in package load(init phase) to only in function call(access phase). Since in the Nginx process environment, only those environment variables exposed through the `env` Nginx config directory can be fetched by `os.getenv` in Lua code(with an exception that env vars can all be fetched in the init phase, and most of the Kong code that using `os.getenv` relies on this). So after #8900, fetching ECS environment variables will fail.

Sorry for bringing this bug in, and this also shows that a full integration test with the AWS environment is really needed for this plugin.

### Full changelog

* Fix fetch ECS environment variable failure caused by the phase change

### Issue reference

Fix [FTI-4340]

[FTI-4340]: https://konghq.atlassian.net/browse/FTI-4340?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by windmgc** [comment]
Manually tested in the following cases:

- EC2 environment, same account lambda invoke
- ECS environment, same account lambda invoke
- EC2 environment, cross account lambda invoke
- ECS environment, cross account lambda invoke

**Comment by windmgc** [comment]
Test added.

File: CHANGELOG.md [hunk 1]
```diff
@@ -78,6 +78,11 @@
   correctly created.
   [#9434](https://github.com/Kong/kong/pull/9434)
 
+#### Plugins
+
+- **AWS Lambda**: Fix issue on fetching environment variables in ECS environment.
```
Comment (dndx) [inline CHANGELOG.md:83]:
```suggestion
- **AWS Lambda**: Fix an issue that is causing inability to read environment variables in ECS environment.
```

**Review by dndx — APPROVED** [review]

**Comment by chan-vince** [comment]
Sorry for the ignorance here, but I think I need this fix but in Kong Konnect.
I've installed the lambda plugin but I cannot see what version of the plugin it uses? Or when a fix to a plugin like this is pushed to master, when it gets added to Konnect?
Thanks

**Comment by windmgc** [comment]
@chan-vince I'm not sure about it, maybe you can reach out the Konnect help for an answer :)

**Comment by rosskukulinski** [comment]
Hi @chan-vince - I'm the overall product lead for Konnect. Once you update your runtimes (Gateway dataplanes) to a version that has this fix included, you should be good-to-go.

**Comment by chan-vince** [comment]
> Hi @chan-vince - I'm the overall product lead for Konnect. Once you update your runtimes (Gateway dataplanes) to a version that has this fix included, you should be good-to-go.

Thanks for jumping in 😀
That is good to know, I sort of thought that might be the case.

My dataplanes are using the `kong/kong-gateway:3.0.0.0` Docker image which I believe is the latest at this moment. I'm guessing that is the same as the `3.0.0` version?

I don't see this PR number in [the release notes](https://github.com/Kong/kong/blob/3.0.0/CHANGELOG.md#300) for any version up and including 3.0.0, so I assume I have to wait a bit.

**Comment by rosskukulinski** [comment]
@chan-vince Since this was just merged 15 days ago, it'll go out in the next gateway release which should be 3.1. It's also been backported to the 2.8.x release branch so it'll go out in the next 2.8 patch release

On the Konnect feedback - would _love_ to hear that. You can reach me at firstname -at- konghq dot com.

**Pull request merged**